tools/build-deb-prepare

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+: "${PACKAGING_BRANCH:=ubuntu/devel}"
+: "${RELEASE:=UNRELEASED}"
+: "${CLOGVER_DEBIAN:=0ubuntu1}"
+: "${DEBFULLNAME:=$(git config user.name)}"
+: "${DEBEMAIL:=$(git config user.email)}"
+
+export DEBFULLNAME DEBEMAIL
+
+fail() { echo "$@" 1>&2; exit 1; }
+
+if [[ "$1" = "-h" || "$1" = "--help" ]]; then
+    cat <<-EOF
+	Usage: ${0##*/}
+	Prepare the source for building a .deb package via e.g. debuild(1).
+	Build a debian/ directory the current git HEAD pulling the debian/
+	directory from branch '$PACKAGING_BRANCH' and updating the changelog,
+	then prepare an .orig tarball matching the new changelog entry.
+	EOF
+    exit
+fi
+
+git clean -f debian || fail "Failed to run git clean."
+
+# This script could work even if a debian/ directory is already existing,
+# but it's a confusing setup that should be avoided.
+[[ ! -e debian ]] || fail "A debian/ directory already exists."
+
+[[ $DEBFULLNAME ]] || fail "Must have DEBFULLNAME or git user.name set."
+[[ $DEBEMAIL ]] || fail "Must have DEBEMAIL or git user.email set."
+
+if [[ $(git status -s) ]]; then
+    cat 1>&2 <<-EOF
+	WARNING: There are uncommitted changes in your working directory.
+	         These changes will not be included in the orig archive.
+	EOF
+fi
+
+# git describe will output something either like '0.1.0' (a tag)
+# or TAG-N-gHASH where N is number of commits since TAG
+uver=$(git describe --abbrev=8) ||
+    fail "Failed to get upstream version with 'git describe'."
+clogver_new="${uver}-${CLOGVER_DEBIAN}"
+
+git fetch "$(git remote)" "$PACKAGING_BRANCH:$PACKAGING_BRANCH" ||
```

Comment:
This fails for me locally, because:

```
$ git remote | wc -l
22
```

Reply:
Heh, you spotted the latest line I added to the script, without checking it well enough apparently :)

The script works fine without that line, but a local copy of the packaging branch has to exist, otherwise we can't `git restore` from it. When cloning the `cloud-init` repo from scratch no local copy of the packaging branches exists. My attempt here was to create a local copy of the packaging branch without doing a `git checkout` roundtrip, e.g.:
```
git checkout ubuntu/focal
git checkout master
```
I couldn't find way to do so, and my attempt was buggy, as you found out. I'm not sure if git allows this.